src/index.ts

```diff
@@ -1,60 +1,114 @@
+import { verify as verifyV3 } from "./3.0/verify";
+import { digestCredential as digestCredentialV3 } from "./3.0/digest";
+import { obfuscateVerifiableCredential as obfuscateVerifiableCredentialV3 } from "./3.0/obfuscate";
 import { OpenAttestationDocument as OpenAttestationDocumentV3 } from "./__generated__/schema.3.0";
-import { obfuscateDocument as obfuscateDocumentV2 } from "./2.0/obfuscate";
-import { obfuscateVerifiableCredential } from "./3.0/obfuscate";
-import { WrapDocumentOptionV2, WrapDocumentOptionV3 } from "./shared/@types/wrap";
-import { SchemaValidationError } from "./shared/utils";
-import { SigningKey, SUPPORTED_SIGNING_ALGORITHM } from "./shared/@types/sign";
-import { ethers, Signer } from "ethers";
-import { getSchema } from "./shared/ajv";
+
+import * as v4 from "./4.0/types";
+import { WrappedDocument as WrappedDocumentV4 } from "./4.0/types";
+import { wrapDocument as wrapDocumentV4, wrapDocuments as wrapDocumentsV4 } from "./4.0/wrap";
+import { signDocument as signDocumentV4 } from "./4.0/sign";
+import { verify as verifyV4 } from "./4.0/verify";
+import { digestCredential as digestCredentialV4 } from "./4.0/digest";
+import { obfuscateVerifiableCredential as obfuscateVerifiableCredentialV4 } from "./4.0/obfuscate";
+import { OpenAttestationDocument as OpenAttestationDocumentV4 } from "./__generated__/schema.4.0";
+
+export function wrapDocument<T extends OpenAttestationDocumentV2>(
+  data: T,
+  options?: WrapDocumentOptionV2
+): WrappedDocumentV2<T> {
+  return wrapDocumentV2(data, { externalSchemaId: options?.externalSchemaId });
+}
+
+export function wrapDocuments<T extends OpenAttestationDocumentV2>(
+  dataArray: T[],
+  options?: WrapDocumentOptionV2
+): WrappedDocumentV2<T>[] {
+  return wrapDocumentsV2(dataArray, { externalSchemaId: options?.externalSchemaId });
+}
 
 export function __unsafe__use__it__at__your__own__risks__wrapDocument<T extends OpenAttestationDocumentV3>(
   data: T,
   options?: WrapDocumentOptionV3
 ): Promise<WrappedDocumentV3<T>> {
-  return wrapV3Document(data, options ?? { version: SchemaId.v3 });
+  return wrapDocumentV3(data, options ?? { version: SchemaId.v3 });
 }
 
 export function __unsafe__use__it__at__your__own__risks__wrapDocuments<T extends OpenAttestationDocumentV3>(
   dataArray: T[],
   options?: WrapDocumentOptionV3
 ): Promise<WrappedDocumentV3<T>[]> {
-  return wrapV3Documents(dataArray, options ?? { version: SchemaId.v3 });
+  return wrapDocumentsV3(dataArray, options ?? { version: SchemaId.v3 });
 }
 
-export function wrapDocument<T extends OpenAttestationDocumentV2>(
+export function _unsafe_use_it_at_your_own_risk_v4_alpha_wrapDocument<T extends OpenAttestationDocumentV4>(
   data: T,
-  options?: WrapDocumentOptionV2
-): WrappedDocumentV2<T> {
-  return wrapV2Document(data, { externalSchemaId: options?.externalSchemaId });
+  options?: WrapDocumentOptionV4
+): Promise<WrappedDocumentV4<T>> {
+  return wrapDocumentV4(data, options ?? { version: SchemaId.v4 });
 }
```

Comment:
version is no longer used, should we remove this? at v4 `wrapDocument` end probably can remove the `options` too?

Reply:
I left the `options` argument there in case we add options down the line. Should we still remove it for now?

Would probably remove the `?? { version: SchemaId.v4 }` portion as it's misleading also